File: backend/internal/models/dice.go
package models

import "time"

// DiceRollRequest representa uma requisição de rolagem de dados
type DiceRollRequest struct {
	Notation   string `json:"notation" binding:"required"` // "2d6+3", "1d20", etc
	Label      string `json:"label,omitempty"`             // Descrição opcional da rolagem
	Advantage  bool   `json:"advantage,omitempty"`         // Rolar com vantagem (2d20, pegar maior)
	Disadvantage bool `json:"disadvantage,omitempty"`      // Rolar com desvantagem (2d20, pegar menor)
}

// DiceRollResponse representa o resultado de uma rolagem de dados
type DiceRollResponse struct {
	Notation     string    `json:"notation"`            // Notação original
	Quantity     int       `json:"quantity"`            // Quantidade de dados
	Sides        int       `json:"sides"`               // Número de lados do dado
	Modifier     int       `json:"modifier"`            // Modificador (+/-)
	Rolls        []int     `json:"rolls"`               // Resultado de cada dado individual
	Total        int       `json:"total"`               // Total da rolagem (soma + modificador)
	Timestamp    time.Time `json:"timestamp"`           // Quando foi rolado
	Label        string    `json:"label,omitempty"`     // Label opcional
	Advantage    bool      `json:"advantage,omitempty"` // Se foi rolado com vantagem
	Disadvantage bool      `json:"disadvantage,omitempty"` // Se foi rolado com desvantagem
	DroppedRolls []int     `json:"dropped_rolls,omitempty"` // Dados descartados (vantagem/desvantagem)
}

// ParsedDice representa a notação de dados parseada
type ParsedDice struct {
	Quantity int // Quantidade de dados
	Sides    int // Lados do dado
	Modifier int // Modificador
}

File: frontend/src/types/dice.ts
// Types for dice rolling system

export type DiceType = 2 | 4 | 6 | 8 | 10 | 12 | 20 | 100;

export interface DiceRoll {
    notation: string;        // "2d6+3"
    rolls: number[];         // [4, 5]
    modifier: number;        // 3
    total: number;           // 12
    timestamp: Date;
    label?: string;          // "Ataque com Espada Longa"
    isCritical?: boolean;    // Natural 20
    isFumble?: boolean;      // Natural 1
    advantage?: boolean;     // Rolado com vantagem
    disadvantage?: boolean;  // Rolado com desvantagem
    droppedRolls?: number[]; // Dados descartados
}

export interface DiceRollRequest {
    notation: string;
    label?: string;
    advantage?: boolean;
    disadvantage?: boolean;
}

export interface DiceRollResponse {
    notation: string;
    quantity: number;
    sides: number;
    modifier: number;
    rolls: number[];
    total: number;
    timestamp: string;
    label?: string;
    advantage?: boolean;
    disadvantage?: boolean;
    dropped_rolls?: number[];
}

export interface QuickRoll {
    label: string;
    notation: string;
    icon?: string;
}
